_includes/aip-head-tag.html

```diff
@@ -25,6 +25,30 @@
   {% for js_script in page.js -%}
   <script type="text/javascript" src="{{ js_script | relative_url }}?v={{ site.github.build_revision }}"></script>
   {% endfor -%}
+  {% if page.aip_index %}
+  <script type="text/javascript">
+    $.when($.ready).then(() => {
+      localStorage.setItem("scope", "{{ page.aip_index.scope }}")
+      console.log(`setting scope to ${localStorage.getItem("scope")}`)
+    })
+  </script>
+  {% else %}
+  <script type="text/javascript">
+    $.when($.ready).then(() => {
+      const content = document.getElementById("aip-main")
+      const anchors = content.getElementsByTagName("a")
+      //console.log(`anchors.length = ${anchors.length}`)
+      let href
+      for(i = 0; i < anchors.length; i++) {
+        href = anchors[i].getAttribute("href")
+        console.log(href)
+      }
+      if(localStorage.getItem("scope")) {
+        console.log(`scope = ${localStorage.getItem("scope")}`)
+      }
+    })
+  </script>
+  {% endif %}
```

Comment:
obviously these need broken out into their own js files.

Using localStorage to store the "context"

I felt like query params were messy. I couldn't use the url path, because if redirecting to a parent AIP, there is no way to retain the original url (i.e. if /custom_org/111 redirects to /111, the end url in the browser is /111). I felt like a cookie was overkill, so I tried this.

Still need to research consequences of using this.

Reply:
Also note, however that you can run locally and try this out. It is printing out the scope to the console. Also not I tried implementing the `Automagic` solution (see below). This is where I felt like it was getting messy and wrote down my ideas below for both feedback and to work out my own thoughts.

Reply:
I talked to a JavaScript friend of mine and apparently localStorage is the new right way to do this kind of thing. I think it is session-based, so it will not persist on a new browser session, which seems fine.